packages/access-client/src/agent.js

```diff
@@ -409,6 +480,115 @@ export class Agent {
     }
   }
 
+  /**
+   * Request authorization of a session allowing this agent to issue UCANs
+   * signed by the passed email address.
+   *
+   * @param {string} email
+   * @param {object} [opts]
+   * @param {AbortSignal} [opts.signal]
+   */
+  async authorize(email, opts) {
+    const sessionPrincipal = emailToSessionPrincipal(email)
+
+    const res = await this.invokeAndExecute(Access.authorize, {
+      audience: this.connection.id,
+      with: this.issuer.did(),
+      nb: {
+        iss: sessionPrincipal.did(),
+        att: [{ can: 'store/*' }, { can: 'provider/add' }],
+      },
+    })
+
+    if (res?.error) {
+      throw new Error('failed to authorize session', { cause: res })
+    }
+
+    const sessionDelegation =
+      /** @type {Ucanto.Delegation<[import('./types').AccessSession]>} */
+      (await this.#waitForDelegation(opts))
+
+    const cap = sessionDelegation.capabilities.find(
+      // @ts-expect-error "key" does not exist in object, unless it's a session capability
+      (c) => c.can === Access.session.can && c.nb.key === this.issuer.did()
+    )
+    if (!cap && isExpired(sessionDelegation)) {
+      throw new Error('received invalid delegation')
+    }
+
+    await this.addProof(sessionDelegation)
+  }
+
+  async claimDelegations() {
+    const res = await this.invokeAndExecute(Access.claim, {
+      audience: this.connection.id,
+      with: this.issuer.did(),
+    })
+
+    if (res.error) {
+      throw new Error('error claiming delegations')
+    }
+    const delegations = Object.values(res.delegations).flatMap((bytes) =>
+      bytesToDelegations(bytes)
```

Comment:
I think you can import this from `@web3-storage/access/encoding` (i.e. this package)

Reply:
(I couldn't in observable due to import error)